src/App.tsx

```diff
@@ -17,18 +63,40 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                onSearch={setQuery}
+                value={selectFilter}
+                setValue={setSelectFilter}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {visibleTodo.length === 0
+
+                ? <Loader />
+
+                : (
+                  <TodoList
+                    todos={visibleTodo}
+                    selectedIdTodo={selectedTodo?.id}
+                    selectedTodo={(idTodo) => {
+                      selectTodo(idTodo);
+                    }}
+                  />
+                )}
+
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal
+          todoUserId={selectedTodo}
+          onClose={(todId: number) => selectTodo(todId)}
```

Comment:
create const for this function and why close function get todoId on close? I hope it could be removed